kernels/volk/volk_32u_reverse_32u.h

```diff
@@ -337,4 +337,57 @@ volk_32u_reverse_32u_arm(uint32_t* out, const uint32_t* in, unsigned int num_poi
 #endif /* LV_HAVE_NEON */
 
 
+#ifdef LV_HAVE_RVV
+#include <riscv_vector.h>
+
+static inline void
+volk_32u_reverse_32u_rvv(uint32_t* out, const uint32_t* in, unsigned int num_points)
+{
+    size_t n = num_points;
+
+    static const uint64_t tblLo[] = {
+        0xE060A020C0408000,
+        0xF070B030D0509010,
+    };
+    static const uint64_t tblHi[] = {
+        0x0E060A020C040800,
+        0x0F070B030D050901,
+    };
+    vuint8m1_t vtblLo = __riscv_vreinterpret_u8m1(__riscv_vle64_v_u64m1(tblLo, 2));
+    vuint8m1_t vtblHi = __riscv_vreinterpret_u8m1(__riscv_vle64_v_u64m1(tblHi, 2));
+
+    size_t vlmax = __riscv_vsetvlmax_e8m1();
+    vuint16m2_t vidx = __riscv_vreinterpret_u16m2(
+        __riscv_vsub(__riscv_vreinterpret_u64m2(__riscv_vid_v_u16m2(vlmax)),
+                     0x3000200010000 - 0x100020003,
+                     vlmax / 4));
+    for (size_t vl; n > 0; n -= vl, in += vl, out += vl) {
+        vl = __riscv_vsetvl_e32m4(n);
+        vuint8m4_t v = __riscv_vreinterpret_u8m4(__riscv_vle32_v_u32m4(in, vl));
+        v = RISCV_PERM4(__riscv_vrgatherei16, v, vidx);
+        vuint8m4_t lo = __riscv_vand(v, 0xF, vl * 4);
+        lo = RISCV_LUT4(__riscv_vrgather, vtblLo, lo);
+        vuint8m4_t hi = __riscv_vsrl(v, 4, vl * 4);
+        hi = RISCV_LUT4(__riscv_vrgather, vtblHi, hi);
+        v = __riscv_vor(hi, lo, vl * 4);
+        __riscv_vse32(out, __riscv_vreinterpret_u32m4(v), vl);
+    }
+}
+#endif /* LV_HAVE_RVV */
+
+#ifdef LV_HAVE_RVA23
+#include <riscv_vector.h>
+
+static inline void
+volk_32u_reverse_32u_rva23(uint32_t* out, const uint32_t* in, unsigned int num_points)
+{
+    size_t n = num_points;
+    for (size_t vl; n > 0; n -= vl, in += vl, out += vl) {
+        vl = __riscv_vsetvl_e32m8(n);
+        vuint32m8_t v = __riscv_vle32_v_u32m8(in, vl);
+        __riscv_vse32(out, __riscv_vbrev(v, vl), vl);
```

Comment:
So `__riscv_vbrev` is an intrinsic for reversal that's introduced with RVA23? Neat.

Reply:
Yes, I recommend using this amazing website as a reference for the intrinsics: https://dzaima.github.io/intrinsics-viewer/